### PR TITLE
Add sort by postal functionality

### PR DESCRIPTION
**Pull request by valerietanhx**
*Sorry branch name is misleading!! Forgot to git checkout :")

Have updated UG as well

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -62,20 +62,20 @@ placeholder
...
-| **Sort**   | `sort [n/] [a/] [d/]`<br> e.g.,`sort d/`                                                                                       |
-| **Import** | `import [FILENAME]`<br> e.g. `import sampledata.csv`                                                                     |
-| **Export** | `export [FILENAME]`<br> e.g. `export sampledata.csv`                                                                     |
-| **Help**   | `help`                                                                                                                   |
-| **Undo**   | `undo [INTEGER]`                                                                                                         |
-| **Redo**   | `redo [INTEGER]`                                                                                                         |
+| **List**   | `list`                                                                                                                                    |
+| **Sort**   | `sort [n/] [a/] [p/] [d/]`<br> e.g.,`sort d/`                                                                                             |
```
Comment (sumhungyee) [inline docs/UserGuide.md:73]: Need to find a way to say at least one

File: docs/UserGuide.md [hunk 2]
```diff
@@ -282,10 +282,10 @@ Format: `list`
 
 Sorts the entire case list based on the specified criteria.
 
-Format: `sort [n/] [a/] [d/]`
+Format: `sort [n/] [a/] [p/] [d/]`
```
Comment (sumhungyee) [inline docs/UserGuide.md:285]: see above comment, how about using or? Naive but probably appropriate
```suggestion
Format: `sort n/ or a/ or p/ or d/`
```

File: src/main/java/seedu/dengue/logic/comparators/PersonPostalComparator.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package seedu.dengue.logic.comparators;
+
+import java.util.Comparator;
+
+import seedu.dengue.model.person.Person;
+
+/**
+ * Comparator for the class {@Person} that compares by its {@Age} attribute
```
Comment (sumhungyee) [inline src/main/java/seedu/dengue/logic/comparators/PersonPostalComparator.java:8]:
```suggestion
 * Comparator for the class {@link Person} that compares by its {@link Postal} attribute
```

File: src/main/java/seedu/dengue/logic/comparators/PostalComparator.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package seedu.dengue.logic.comparators;
+
+import java.util.Comparator;
+
+import seedu.dengue.model.person.Postal;
+
+/**
+ * Comparator for the class {@code Age}
```
Comment (sumhungyee) [inline src/main/java/seedu/dengue/logic/comparators/PostalComparator.java:8]:
```suggestion
 * Comparator for the class {@code Postal}
```

File: src/main/java/seedu/dengue/logic/comparators/PostalComparator.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+     */
+    @Override
+    public int compare(Postal p1, Postal p2) {
+        return p1.value.compareTo(p2.value);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        return this == obj;
+    }
```
Comment (sumhungyee) [inline src/main/java/seedu/dengue/logic/comparators/PostalComparator.java:27]: Will never be used but might want to use the long instanceof whatever. I don't think it's necessary.

**Review by sumhungyee — CHANGES_REQUESTED** [review]
LGTM, just fix a bit of javadocs! :D

File: docs/UserGuide.md (hunk 1 above)
Comment (valerietanhx) [reply docs/UserGuide.md:73]: Agree, need to work it out with the rest of the team!! Currently seem to be settling on using {} but not confirmed yet; can update in a future PR

File: docs/UserGuide.md (hunk 2 above)
Comment (valerietanhx) [reply docs/UserGuide.md:285]: See above comment as well re: using {} instead! Can bring up with everyone

File: src/main/java/seedu/dengue/logic/comparators/PostalComparator.java (hunk 5 above)
Comment (valerietanhx) [reply src/main/java/seedu/dengue/logic/comparators/PostalComparator.java:27]: Mm idt it's necessary either ba, can leave it for now!

**Review by sumhungyee — APPROVED** [review]
LGTM

**Pull request merged**